src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -92,6 +93,7 @@ public partial struct JsonDocumentOptions
         public bool TryGetUInt32(out uint value) { throw null; }
         [System.CLSCompliantAttribute(false)]
         public bool TryGetUInt64(out ulong value) { throw null; }
+        public static bool TryParseValue(ref System.Text.Json.Utf8JsonReader reader, [System.Diagnostics.CodeAnalysis.NotNullWhenAttribute(true)] out System.Text.Json.JsonElement? element) { throw null; }
```

Comment:
`JsonElement` is a struct. What does `NotNullWhen(true)` attribute mean for value types like this?

I would have thought we wouldn't need it.
```suggestion
        public static bool TryParseValue(ref System.Text.Json.Utf8JsonReader reader, out System.Text.Json.JsonElement? element) { throw null; }
```

Reply:
TryParseValue's out param returns `Nullable<JsonElement>` so we need the attribute.  Using `Nullable` instead of `default(JsonElement)` prevents any possible misuse of `default(JsonElement)` and perhaps a bit less on the stack.

Reply:
Gotcha. That makes more sense now :) Thanks.